go/private/rules/library.bzl

```diff
@@ -204,8 +207,37 @@ go_library = rule(
 
 # See docs/go/core/rules.md#go_library for full documentation.
 
+def _go_tool_library_impl(ctx):
+    """Implements the go_tool_library() rule."""
+    go = go_context(ctx, include_deprecated_properties = False)
+
+    library = go.new_library(go)
+    source = go.library_to_source(go, ctx.attr, library, ctx.coverage_instrumented())
+    archive = go.archive(go, source)
+    validation_output = archive.data._validation_output
+
+    return [
+        library,
+        source,
+        archive,
+        DefaultInfo(
+            files = depset([archive.data.export_file]),
+        ),
+        coverage_common.instrumented_files_info(
```

Comment:
I split this since it has a different set of attirbutes. this rule is only really used for `go/tools/bzltestutil/chdir`, right? Do we actually need the coverage provider or the OutputGroupInfo?